backend/saves.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+from dataclasses import dataclass
+from sqlite3 import connect
+import json
+
+@dataclass
+class Saves():
+#
+    DBPath: str
+#
+    RMTABLE = 'drop table if exists games'
+    MKTABLE = 'create table games (id varchar(3) primary key, data json)'
+#
+    GET_ALL = 'SELECT id, json_extract(data, "$") as data from games'
+    GET_ID = 'SELECT id, data from games where id = ? '
+#
+    DELETE = 'DELETE FROM games WHERE id = ?'                      
+    UPDATE = 'INSERT OR REPLACE INTO games values(?, ?) '
+#
+
+    def _dictate(self, cursor, row):
+        fields = [column[0] for column in cursor.description] 
+        return {key: value for key, value in zip(fields, row)}
+
+    def _connect(self):
+        conn = connect(self.DBPath)
+        conn.row_factory = self._dictate
+        return conn
+
+    def _initdb(self):
+        c = self._connect()
+        c.execute(self.RMTABLE)
+        c.execute(self.MKTABLE)
+        c.close()
+    
+    def get_all(self):
```

Comment:
Since a module is basically a singleton we could pull these functions into `database.py` as functions. Unless you were thinking about something subclassing `Save`?

Reply:
done